dask/bag/core.py

```diff
@@ -2294,43 +2294,48 @@ def groupby_tasks(b, grouper, hash=hash, max_branch=32):
 
     token = tokenize(b, grouper, hash, max_branch)
 
+    shuffle_join_token = "shuffle-join-" + token
+
     start = dict(
-        (("shuffle-join-" + token, 0, inp), (b2.name, i) if i < b.npartitions else [])
```

Comment:
These  changes look good @galipremsagar -  Although, I do have a minor preference for names like `shuffle_join_name` rather than `..._token` (this is proabably just a personal preference).